src/devices/Board/RaspberryPiBoard.cs

```diff
@@ -19,10 +24,70 @@ namespace Iot.Device.Board
     public class RaspberryPiBoard : GenericBoard
     {
         private readonly object _initLock = new object();
+        private readonly string[] _possibleI2cActivations = new string[]
+        {
+            "dtparam=i2c=on",
+            "dtparam=i2c_arm=on",
+            "dtparam=i2c_baudrate=",
+            "dtparam=i2c_arm_baudrate=",
+            // This is taking the assumption that any other parameter will go after.
+            // Those parameters are almost never used.
+            "dtoverlay=i2c0,pins_0_1",
+            "dtoverlay=i2c0,pins_28_29",
+            "dtoverlay=i2c0,pins_44_45",
+            "dtoverlay=i2c0,pins_46_47",
+            "dtoverlay=i2c1,pins_2_3",
+            "dtoverlay=i2c1,pins_44_45",
+            "dtoverlay=i2c3,pins_2_3",
+            "dtoverlay=i2c3,pins_4_5",
+            "dtoverlay=i2c4,pins_6_7",
+            "dtoverlay=i2c4,pins_8_9",
+            "dtoverlay=i2c5,pins_10_11",
+            "dtoverlay=i2c5,pins_12_13",
+            "dtoverlay=i2c6,pins_0_1",
+            "dtoverlay=i2c6,pins_22_23",
```

Comment:
If I read the documentation correctly, the parameter is optional. So it should be sufficient to test for `dtoverlay=i2c0`, `dtoverlay=i2c1` etc.

Reply:
Let me test all this. I'll adjust like for SPI if not mandatory.